assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -0,0 +1,217 @@
+import { HighlightMatches } from '../HighlightMatches';
+
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+const { __ } = wp.i18n;
+
+const placeholderData = {
+  header: [ 'Lorem', 'Ipsum', 'Dolor' ],
+  rows: [
+    [ 'Lorem', 'Ipsum', 'Dolor' ],
+    [ 'Sit', 'Amet', 'Lorem' ],
+    [ 'Amet', 'Ipsum', 'Sit' ],
+  ]
+};
+
+export class SpreadsheetFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: null,
+      spreadSheetData: null,
+      searchText: '',
+      sortDirection: 'asc',
+      sortColumnIndex: null,
+    };
+
+    this.onHeaderClick = this.onHeaderClick.bind(this);
+  }
+
+  onHeaderClick( index ) {
+    if ( index == this.state.sortColumnIndex ) {
+      const newDirection = this.state.sortDirection == 'asc' ? 'desc' : 'asc';
+      this.setState({
+        sortDirection: newDirection
+      });
+    } else {
+      this.setState({
+        sortColumnIndex: index,
+        sortDirection: 'asc',
+      });
+    }
+  }
+
+  fetchSheetData( url ) {
+    const sheetID = this.extractSheetID( url );
+
+    if (sheetID !== false) {
+      if ( this.props.handleErrors ) {
+        this.props.handleErrors({ invalidSheetId: false })
+      }
+      this.setState({ loading: true });
+
+      const queryArgs = {
+        path: addQueryArgs('/planet4/v1/get-spreadsheet-data', {
+          sheet_id: sheetID,
+        })
+      };
+
+      apiFetch( queryArgs )
+        .then( spreadSheetData => {
+          this.setState( {
+            loading: false,
+            spreadSheetData
+          } );
+      } );
+    } else {
+      if ( this.props.handleErrors ) {
+        this.props.handleErrors( { invalidSheetId: true } );
+      }
+      this.setState({ loading: false, spreadSheetData: null });
+    }
+  }
+
+  componentDidMount() {
+    if ( this.props.url != '' ) {
+      this.fetchSheetData(this.props.url);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.url !== this.props.url) {
+      this.fetchSheetData(nextProps.url);
+    }
+  }
+
+  extractSheetID( url ) {
+    const googleSheetsPattern = /https:\/\/docs\.google\.com\/spreadsheets\/d\/e\/([\w-]+)/;
+    const matches = url.match(googleSheetsPattern);
+    if (matches !== null) {
+      return matches[1];
+    } else {
+      return false;
+    }
+  }
+
+  sortRows( rows, columnIndex ) {
+    if (columnIndex === null) {
+      return rows;
+    }
+
+    const sortedRows = rows.sort( ( rowA, rowB ) => {
+      const textCompare = rowA[columnIndex].localeCompare( rowB[columnIndex] );
+      if ( textCompare !== 0 ) {
+        return textCompare;
+      }
+    } );
+    if (this.state.sortDirection === 'desc') {
+      return sortedRows.reverse();
+    } else {
+      return sortedRows;
+    }
+  };
+
+  filterMatchingRows( rows ) {
+    const filteredRows = rows.filter( row => {
+      return !!row.find( cell => cell.toLowerCase().includes( this.state.searchText.trim().toLowerCase() ) );
```

Comment:
You can use `row.some` instead here, which already returns a boolean for you.

Reply:
Neat 💯